test/e2e/Maker.t.sol

```diff
@@ -0,0 +1,161 @@
+        prepareSolverWrapper();
+    }
+
+    function prepareSolverWrapper() private {
+        solverWrapper = new ERC3156FlashLoanSolverWrapper(Constants.SETTLEMENT_CONTRACT);
+
+        // The solver wrapper must be a solver because it directly calls
+        // `settle`.
+        CowProtocol.addSolver(vm, address(solverWrapper));
+
+        // Call `approve` from the settlement contract so that DAI can be spent
+        // on a settlement interaction on behalf of the solver wrapper. With an
+        // unlimited approval, this step only needs to be performed once per
+        // loaned token.
+        ICowSettlement.Interaction[] memory onlyApprove = new ICowSettlement.Interaction[](1);
+        onlyApprove[0] = interactionWrapperDaiApprove(address(Constants.SETTLEMENT_CONTRACT), type(uint256).max);
+        vm.prank(solver);
+        CowProtocol.emptySettleWithInteractions(onlyApprove);
+    }
+
+    function test_settleWithFlashLoan() external {
+        uint256 loanedAmount = 10_000 ether; // $10,000
+
+        uint256 flashFee = MAKER_FLASH_LOAN_CONTRACT.flashFee(address(Constants.DAI), loanedAmount);
+        // Flash loan fees are always zero in the Maker contract. We just need
+        // to repay the borrowed collateral at the end.
+        // <https://etherscan.io/address/0x07df2ad9878F8797B4055230bbAE5C808b8259b3#code#F1#L31>
+        assertEq(flashFee, 0);
+
+        uint256 settlementInitialDaiBalance = Constants.DAI.balanceOf(address(Constants.SETTLEMENT_CONTRACT));
+        TokenBalanceAccumulator.Balance[] memory expectedBalances = new TokenBalanceAccumulator.Balance[](3);
+
+        // Start preparing the settlement interactions.
+        ICowSettlement.Interaction[] memory interactionsWithFlashLoan = new ICowSettlement.Interaction[](6);
+        // First, we confirm that, at the point in time of the settlement, the
+        // flash loan proceeds are indeed stored in the wrapper solver.
+        interactionsWithFlashLoan[0] = interactionPushDaiBalanceToAccumulator(address(solverWrapper));
+        expectedBalances[0] = TokenBalanceAccumulator.Balance(Constants.DAI, address(solverWrapper), loanedAmount);
+        // Second, we double check that the settlement balance hasn't changed.
+        interactionsWithFlashLoan[1] = interactionPushDaiBalanceToAccumulator(address(Constants.SETTLEMENT_CONTRACT));
+        expectedBalances[1] = TokenBalanceAccumulator.Balance(
+            Constants.DAI, address(Constants.SETTLEMENT_CONTRACT), settlementInitialDaiBalance
+        );
+        // Third, we make sure we can transfer these tokens. We do that by
+        // trying a transfer into the settlement contract. The expectation is
+        // that the wrapper has already approved the settlement contract to
+        // transfer its tokens out of it. In practice, the target of the
+        // transfer is expected to be the user rather than the settlement
+        // contract for gas efficiency.
```

Comment:
Why would transferring to user be more efficient in practice, when funds are supposed to flow from user to settlement contract anyway?